typed/typed.go

```diff
@@ -32,6 +32,14 @@ const (
 	AllowDuplicates ValidationOptions = iota
 )
 
+// ExtractItemsOptions is the list of all the options available when extracting items.
+type ExtractItemsOptions int
+
+const (
+	// AppendKeyFields means that when extracting items, the key field would also be included.
+	AppendKeyFields ExtractItemsOptions = iota
+)
+
```

Comment:
Are we confident all future options will be flags?  Alternative form that would handle arbitrary option params: https://github.com/kubernetes/kube-openapi/pull/519/files#diff-72f8260914152fe9557b05d756ec94854184645e991ff755855628428d6d7a18R410-R421

Reply:
SG, updated using this format.